### PR TITLE
Improve ping and reconnect logic

### PR DESCRIPTION
**Pull request by apop880**
#203 should be resolved by this. I've cleaned up the logic around pinging to use the standard ping and pong options provided by the Home Assistant WebSocket interface, and I added some additional checks to ensure that pings don't continue to be sent if the connection is stopping or starting. I've thoroughly tested and this has resolved reconnection issues I was having.

### CODE REVIEW

File: scripts/controllers/main.js [hunk 1]
```diff
@@ -1931,20 +1931,21 @@ function MainController ($scope, $location) {
...
 
-      Api.getUser(function (res) {
+      Api.sendPing(function (res) {
          if('id' in res) success = true;
       });
 
       setTimeout(function () {
          if(success) return;
 
+         realReadyState = false;
```
Comment (Evgeny-) [inline scripts/controllers/main.js]: Why does it needed?

File: README.md [hunk 2]
```diff
@@ -67,6 +67,12 @@ var CONFIG = {
     */
    authToken: null,
 
+   /* pingMaxTimeout: Set to false disable pinging of the websocket connection.
+    * Otherwise, a ping will be sent every five seconds, and if a response is not received in 3 seconds,
+    * a reconnect will be attempted. If not included in the config file, setting defaults to true.
+    */
+   pingMaxTimeout: true,
```
Comment (Evgeny-) [inline README.md]: Seems a little bit obsolete name, since it's not about "max timeout"

**Comment by Evgeny-** [comment]
Thank you for that. I've added few comments on questionable lines, can you please comment them?

**Comment by Evgeny-** [comment]
Thank you for that. Should we change `pingMaxTimeout` name?

File: README.md (hunk 2 above)
Comment (apop880) [reply README.md]: I updated ```pingMaxTimeout``` to ```pingConnection```, I agree that makes more sense.

**Comment by apop880** [comment]
Pushed a new commit with ```pingMaxTimeout``` changed to ```pingConnection``` and an existing reference to ```pingMaxTimeout``` that I didn't catch before removed from the readme. Let me know if you have any other questions.

**Comment by Evgeny-** [comment]
Everything seems good. Thank you

**Pull request merged**